frontend/src/views/login/components/login-form.vue

```diff
@@ -205,11 +205,25 @@ const loginForm = reactive({
 });
 
 const loginRules = reactive({
-    name: [{ required: true, message: i18n.t('commons.rule.username'), trigger: 'blur' }],
-    password: [{ required: true, message: i18n.t('commons.rule.password'), trigger: 'blur' }],
+    name: [{ required: true, validator: checkUsername, trigger: 'blur' }],
+    password: [{ required: true, validator: checkPassword, trigger: 'blur' }],
     agreeLicense: [Rules.requiredSelect, { type: 'array', validator: checkAgreeLicense, trigger: 'blur' }],
 });
 
+function checkUsername(rule: any, value: any, callback: any) {
+    if (value === '') {
+        return callback(new Error(i18n.t('commons.rule.username')));
+    }
+    callback();
+}
+
+function checkPassword(rule: any, value: any, callback: any) {
+    if (value === '') {
+        return callback(new Error(i18n.t('commons.rule.password')));
+    }
+    callback();
+}
+
 function checkAgreeLicense(rule: any, value: any, callback: any) {
     if (!value && !_isMobile()) {
         return callback(new Error(i18n.t('commons.login.errorAgree')));
```

Comment:
Your code looks generally clean and efficient. Here are some observations and potential improvements:

1. **Naming Consistency**: The `checkUsername` and `checkPassword` functions use underscores (`_`) instead of double underscores to denote private methods, which is consistent with most programming standards.

2. **Error Messages**: There's nothing inherently wrong with using inline strings for error messages like "i18n.t('commons.rule.username')", but you might consider creating constants for these values to keep your code cleaner and more maintainable.

3. **Function Comments**: It would be helpful to add comments to explain what each function is doing. For example, in `checkAgreeLicense`, adding a comment about why `_isMobile()` is used might clarify its purpose.

4. **Type Annotations**: While TypeScript annotations have been omitted here to focus on syntax, they can help catch errors during runtime and improve code readability. Consider adding type annotations where appropriate.

Here's an improved version with additional comments:

```javascript
const loginForm = reactive({
});

const loginRules = reactive({
    name: [
        {required: true, validator: checkUsername, message: i18n.t('commons.rule.username'), trigger: 'blur'},
    ],
    password: [
        {required: true, validator: checkPassword, message: i18n.t('commons.rule.password'), trigger: 'blur'},
    ],
    agreeLicense: [
        Rules.requiredSelect,
        {type: 'array', validator: checkAgreeLicense, message: i18n.t('commons.login.errorAgree', { _mobileMessage: _isMobile() ? 'Cannot accept license terms.' : '' }), trigger: 'blur'}
    ]
});

/**
 * Validates the username input.
 */
function checkUsername(rule: any, value: any, callback: any): void {
    if (value === '') {
        return callback(new Error(i18n.t('commons.rule.username')));
    }
    callback();
}

/**
 * Validates the password input.
 */
function checkPassword(rule: any, value: any, callback: any): void {
    if (value === '') {
        return callback(new Error(i18n.t('commons.rule.password')));
    }
    callback();
}

/**
 * Validates whether user has agreed to the terms.
 * If necessary, provides a custom message considering device type.
 */
function checkAgreeLicense(rule: any, value: any, callback: any): void {
    // Check if there's no agreement, indicating lack of acceptance
    if (!value && !_isMobile()) {
        // Provide a specific error message based on mobile status
        return callback(new Error(i18n.t('commons.login.errorAgree', { _mobileMessage: _isMobile() ? 'Cannot accept license terms.' : '' }));
    }
    callback();
}
```

These changes make the code easier to understand and potentially reduce bugs by providing clear explanations within the function definitions.